src/Files.App/Views/WidgetsPage.xaml.cs

```diff
@@ -133,39 +133,25 @@ private async void RecentFilesWidget_RecentFileInvoked(object sender, UserContro
 		{
 			try
 			{
-				var directoryName = Path.GetDirectoryName(e.ItemPath);
-				await Win32Helpers.InvokeWin32ComponentAsync(e.ItemPath, AppInstance, workingDirectory: directoryName);
-			}
-			catch (UnauthorizedAccessException)
-			{
-				DynamicDialog dialog = DynamicDialogFactory.GetFor_ConsentDialog();
-				await SetContentDialogRoot(dialog).ShowAsync();
-			}
-			catch (ArgumentException)
-			{
-				if (new DirectoryInfo(e.ItemPath).Root.ToString().Contains(@"C:\", StringComparison.Ordinal))
+				if (e.IsFile)
 				{
-					AppInstance.NavigateWithArguments(FolderSettings.GetLayoutType(e.ItemPath), new NavigationArguments()
-					{
-						NavPathParam = e.ItemPath
-					});
+					var directoryName = Path.GetDirectoryName(e.ItemPath);
+					await Win32Helpers.InvokeWin32ComponentAsync(e.ItemPath, AppInstance, workingDirectory: directoryName);
 				}
 				else
 				{
-					foreach (DriveItem drive in Enumerable.Concat(App.DrivesManager.Drives, App.CloudDrivesManager.Drives))
+					AppInstance.NavigateWithArguments(FolderSettings.GetLayoutType(e.ItemPath), new NavigationArguments()
```

Comment:
Fix: when opening a folder from recent items, navigate to it inside Files instead of opening it in explorer. Code path currently unused.